src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -415,12 +413,16 @@ private static KeyExchangeProperties chooseCipherSuite(
             }
 
             List<CipherSuite> legacySuites = new LinkedList<>();
+            boolean CSFound = false;
             for (CipherSuite cs : preferred) {
                 if (!HandshakeContext.isNegotiable(
                         proposed, shc.negotiatedProtocol, cs)) {
                     continue;
```

Comment:
I may add a debug log that the cipher suite is not negotiable her for the protocol.

Reply:
If there is no common cipher suite between the client and server. All of the server cipher suites will be printed out. The error message of "no common cipher suites" will be logged. If any cipher suites are shared between the client and server, the rest of the cipher suites would not be used. For this case, the error message of "key exchange failed" will be logged.